pyanaconda/payload/dnf/payload.py

```diff
@@ -141,7 +141,8 @@ def set_from_opts(self, opts):
         """
         # Set the source based on opts.method if it isn't already set
         # - opts.method is currently set by command line/boot options
-        if opts.method and (not self.proxy.Sources or self.source_type == DNF_DEFAULT_SOURCE_TYPE):
+        current_source_is_default = self.source_type == conf.payload.default_source
```

Comment:
The property `self.source_type` will create a default source if there is none. You don't really want that in this case. Please, move this code back to the condition or create a new method `is_source_default` and call it AFTER we check `opts.method` and `not self.proxy.Sources`.

Reply:
Sounds good, moving to a separate method.